clientconfig/tracing.go

```diff
@@ -0,0 +1,163 @@
+package clientconfig
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var defaultSamplingProbability = 0.1
+
+// propagator to use
+var propagator propagation.TextMapPropagator = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}) // traceparent header
+type Option interface {
+	apply(*options)
+}
+type options struct {
+	address string
+}
+
+// WithAddress takes an address in the form of Host:Port
+func WithAddress(addr string) Option {
+	return addrOption{address: addr}
+}
+
+type addrOption struct {
+	address string
+}
+
+func (o addrOption) apply(opts *options) {
+	opts.address = o.address
+}
+
+// OtlpGrpcExporter uses the otlptracegrpc modules and the otlptrace module to produce a new exporter at our default addr
+func OtlpGrpcExporter(ctx context.Context, opts ...Option) (sdktrace.SpanExporter, error) {
+
+	const DefaultCollectorHost = "localhost"
+	const defaultCollectorPort uint16 = 4317
+	addr := fmt.Sprintf("%s:%d", DefaultCollectorHost, defaultCollectorPort)
+
+	options := options{
+		address: addr,
+	}
+
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	otlpClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithEndpoint(addr), //Not strictly needed as we use the defaults
+		otlptracegrpc.WithReconnectionPeriod(15*time.Second),
+		otlptracegrpc.WithInsecure(),
+	)
+
+	spanExporter, err := otlptrace.New(ctx, otlpClient)
+	if err != nil {
+		return nil, err
+	}
+	return spanExporter, nil
+
+}
+
+// DefaultInstrumentor returns the transport to use in client requests.
+// It takes in a transport to wrap, e.g. http.DefaultTransport, and the request
+// context value to pull the span name out from.
+// 99% sure this is wrapping a wrapped thing and totally redundant. Fix later.
+func DefaultInstrumentor(spanNameCtxValue interface{}, baseTransport http.RoundTripper, tp sdktrace.TracerProvider) http.RoundTripper {
```

Comment:
does `spanNameCtxValue` need to be an interface instead of `context.Context`?

Reply:
Same comment for line 85

Reply:
Love that we can drop `ctx` then from `WithTracing`! 

I'm going to push a commit that does what I think you suggested, but I'm not sure so feel free to correct it